plugins/modules/mongodb_schema.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Rhys Campbell (@rhysmeister) <rhyscampbell@bluewin.ch>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: mongodb_schema
+
+short_description: Manages MongoDB Document Schema Validators.
+
+description:
+- Manages MongoDB Document Schema Validators.
+- Create, update and remove Validators on a collection.
+- Supports the entire range of jsonSchema keywords.
+- See [jsonSchema Available Keywords](https://docs.mongodb.com/manual/reference/operator/query/jsonSchema/#available-keywords) for details.
+
+author: Rhys Campbell (@rhysmeister)
+version_added: "1.3.0"
+
+extends_documentation_fragment:
+  - community.mongodb.login_options
+  - community.mongodb.ssl_options
+
+options:
+  db:
+    description:
+      - The database to work with.
+    required: yes
+    type: str
+  collection:
+    description:
+      - The collection to work with.
+    required: yes
+    type: str
+  bsonType:
+    description:
+      - Expected document type.
+    type: str
+    default: "object"
```

Comment:
Can this every be something other than `object`? As I understand MongoDB documents are always objects. Maybe we should just hard code this to be `object`.

Reply:
@cognifloyd Can't find anything confirming this but you're almost certainly right.